power_grid_model_c/power_grid_model/include/power_grid_model/component/power_sensor.hpp

```diff
@@ -86,15 +107,31 @@ template <bool sym> class PowerSensor : public GenericPowerSensor {
         s_measured_ = ps + 1.0i * qs;
     }
 
-    SensorCalcParam<true> sym_calc_param() const final {
-        SensorCalcParam<true> calc_param{};
-        calc_param.variance = power_sigma_ * power_sigma_;
+    PowerSensorCalcParam<true> sym_calc_param() const final {
+        PowerSensorCalcParam<true> calc_param{};
+        if (is_normal(p_sigma_) && is_normal(q_sigma_)) {
+            calc_param.p_variance = mean_val(p_sigma_ * p_sigma_);
+            calc_param.q_variance = mean_val(q_sigma_ * q_sigma_);
+        } else {
+            auto const variance = is_nan(p_sigma_) ? apparent_power_sigma_ * apparent_power_sigma_ / 2
+                                                   : std::numeric_limits<double>::infinity();
+            calc_param.p_variance = variance;
+            calc_param.q_variance = variance;
+        }
         calc_param.value = mean_val(s_measured_);
         return calc_param;
     }
-    SensorCalcParam<false> asym_calc_param() const final {
-        SensorCalcParam<false> calc_param{};
-        calc_param.variance = power_sigma_ * power_sigma_;
+    PowerSensorCalcParam<false> asym_calc_param() const final {
+        PowerSensorCalcParam<false> calc_param{};
+        if (is_normal(p_sigma_) && is_normal(q_sigma_)) {
+            calc_param.p_variance = piecewise_real_value(p_sigma_ * p_sigma_);
+            calc_param.q_variance = piecewise_real_value(q_sigma_ * q_sigma_);
+        } else {
+            auto const variance = RealValue<false>{is_nan(p_sigma_) ? apparent_power_sigma_ * apparent_power_sigma_ / 2
+                                                                    : std::numeric_limits<double>::infinity()};
+            calc_param.p_variance = variance;
+            calc_param.q_variance = variance;
+        }
         calc_param.value = piecewise_complex_value(s_measured_);
```

Comment:
add a debug assertion here to check if `p_sigma` and `q_sigma` are set with one NaN and one not NaN, as it is undefined.